tests/components/habitica/test_util.py

```diff
@@ -0,0 +1,177 @@
+"""Tests for Habitica utility functions."""
+
+from datetime import date, datetime
+from typing import Any
+
+import pytest
+
+from homeassistant.components.habitica.util import next_due_date, to_date
```

Comment:
Typically home assistant code is only tested through the home assistant core APIs and helper modules like this are not united tested. The options are to move this code into the library or to let this be tested through the other tests.

(Appreciate the intent to have high test coverage of the corner cases)

Reply:
I had been requested in other integrations to test utility functions individually so I'm doing the same here. Moving code to the library is planned for a new library but it is not finished yet, so this has to stay for now.

Reply:
I can't remember exactly when and in what PR, but ever since i'm adding tests for helper functions in a `test_util.py` file, the bring integration and ista_ecotrend for example have a `test_util.py`. I think it was @joostlek who suggested doing it like that.

Reply:
It would not surprise me if I did say this. But I kinda do agree that we should ideally try to test this on where this is used, since otherwise if the usage of the util changes, you might fall into the case where you maintain code just for the tests.

Reply:
I can also understand that there are cases where it is extremely tedious to test since you have to put in whole new fixtures for just touching that one line. But I do think that that is the best way to test, since it will also tell the dev breaking them "This is the use case.", and with separate unit tests for a helper it can be hard to see

Reply:
I think the to_date function will be superfluous soon, as it will be part of the dataclass serialization in the new library I'mm working on (and also it was fixed in the Habitica API some weeks ago), I could remove it for now if that is ok. For the next_due_date function i'll move it to `test_todo.py` were it maybe fits better and see if i can cover it with fixtures.